lib/Zonemaster/Backend/Config.pm

```diff
@@ -29,29 +30,152 @@ Readonly my @SIG_NAME => split ' ', $Config{sig_name};
 =cut
 
 sub load_config {
-    my ( $class, $params ) = @_;
-    my $self = {};
+    my ( $class ) = @_;
 
     $log->notice( "Loading config: $path" );
+    my $text = read_file $path;
+
+    my $obj = eval { $class->parse( $text ) };
+    if ( $@ ) {
+        die "File $path: $@";
+    }
+
+    return $obj;
+}
+
+=head2 parse
+
+Parse a new Zonemaster::Backend::Config from the contents of a
+L<configuration|https://github.com/zonemaster/zonemaster-backend/blob/master/docs/Configuration.md>
+file.
+
+    my $config = Zonemaster::Backend::Config->parse(
+        q{
+            [DB]
+            engine = SQLite
+
+            [SQLITE]
+            database_file = /var/db/zonemaster.sqlite
+        }
+    );
+
+Throws an exception if the given configuration file contains errors.
```

Comment:
I think it makes sense to state, "Undefined sections and keys will be silently ignored", or something similar.

Reply:
Any comment here?

Reply:
Yeah, I guess parse() isn't much of a goal keeper at this point. Though it does detect a few errors.

If you insist I'll change the wording now and then change it back in the next PR. (Though the validation layer won't be complete until the PR after that.)

Reply:
Yes include that, and I'll approve.

Reply:
I added a note on this.